src/middlewared/middlewared/utils/smb.py

```diff
@@ -107,7 +107,7 @@ class SMBSharePurpose(enum.StrEnum):
     '%', '<', '>', '*', '?', '|', '/', '\\', '+', '=', ';', ':', '"', ',', '[', ']'
 })
 RESERVED_SHARE_NAMES = frozenset({'global', 'printers', 'homes', 'admin$', 'ipc$'})
-SUPPORTED_SMB_VARIABLES = frozenset({'U', 'u', 'G'})  # see man 5 smb.conf "VARIABLE SUBSTITUTIONS"
+SUPPORTED_SMB_VARIABLES = frozenset({'U', 'u', 'G', 'D'})  # see man 5 smb.conf "VARIABLE SUBSTITUTIONS"
```

Comment:
A question: do we really want to allow lowercase u?

FWIW, I do see "%u" being used in `test__change_sharing_smd_home_to_true` in _test_420_smb.py_.